### PR TITLE
deploy: pin retries to a const and forget correctly in the dc loop

### PR DESCRIPTION
**Pull request by kargakis**
@deads2k one more for today :)

### CODE REVIEW

File: pkg/deploy/controller/deploymentconfig/controller.go [hunk 1]
```diff
@@ -375,7 +376,7 @@ func (c *DeploymentConfigController) handleErr(err error, key interface{}) {
 		return
 	}
 
-	if c.queue.NumRequeues(key) < 10 {
+	if c.queue.NumRequeues(key) < MaxRetries {
 		c.queue.AddRateLimited(key)
 	} else {
 		glog.V(2).Infof(err.Error())
```
Comment (deads2k) [inline pkg/deploy/controller/deploymentconfig/controller.go]: I prefer your upstream structure with the reporting on retry, return, then util.HandleErr, forget.

File: pkg/deploy/controller/generictrigger/factory.go [hunk 2]
```diff
@@ -163,7 +166,7 @@ func (c *DeploymentTriggerController) work() bool {
 	if err := c.Handle(dc); err != nil {
 		utilruntime.HandleError(err)
 
-		if c.queue.NumRequeues(key) < 2 {
+		if c.queue.NumRequeues(key) < MaxRetries {
```
Comment (deads2k) [inline pkg/deploy/controller/generictrigger/factory.go]: Any reason we should use your `handleErr` style function from upstream?

**Comment by kargakis** [comment]
@deads2k comments addressed, ptal

**Comment by kargakis** [comment]
[test]

**Comment by deads2k** [comment]
lgtm

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/9767

[merge]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**